applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -123,17 +101,9 @@ public static ConfigurationAsyncClientBuilder builder() {
      * @throws ResourceModifiedException If a ConfigurationSetting with the same key and label exists.
      * @throws HttpResponseException If {@code key} is an empty string.
      */
-    public Mono<Response<ConfigurationSetting>> addSetting(ConfigurationSetting setting) {
-        // Validate that setting and key is not null. The key is used in the service URL so it cannot be null.
-        validateSetting(setting);
-
-        // This service method call is similar to setSetting except we're passing If-Not-Match = "*". If the service
-        // finds any existing configuration settings, then its e-tag will match and the service will return an error.
-        return service.setKey(serviceEndpoint, setting.key(), setting.label(), setting, null, getETagValue(ETAG_ANY))
-            .doOnRequest(ignoredValue -> logger.asInformational().log("Adding ConfigurationSetting - {}", setting))
-            .doOnSuccess(response -> logger.asInformational().log("Added ConfigurationSetting - {}", response.value()))
-            .onErrorMap(ConfigurationAsyncClient::addSettingExceptionMapper)
-            .doOnError(error -> logger.asWarning().log("Failed to add ConfigurationSetting - {}", setting, error));
+    public Mono<ConfigurationSetting> addSetting(ConfigurationSetting setting) {
+        return client.addSetting(setting)
+            .flatMap(response -> Mono.justOrEmpty(response.value()));
```

Comment:
If we take this approach this code pattern will become so common that we will want to optimise it. I propose you create some static utility method named `unwrapResponse` or something that would be called along the lines of `return unwrapResponse(client.addSetting(setting))`